peekingduck/pipeline/nodes/output/screen.py

```diff
@@ -49,15 +51,33 @@ def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
         cv2.namedWindow(self.window_name, cv2.WINDOW_NORMAL)
         cv2.moveWindow(self.window_name, self.window_loc["x"], self.window_loc["y"])
+        self.aspect_ratio_thres = 5e-3
+        self.current_filename = None
 
     def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         """Show the outputs on your display"""
-
         img = inputs["img"]
-        if self.window_size["do_resizing"]:
-            img = cv2.resize(
-                img, (self.window_size["width"], self.window_size["height"])
+        if not self.window_size["do_resizing"]:
+            img_height, img_width, _ = img.shape
+            if inputs["filename"] is not self.current_filename:
```

Comment:
Use `==` or `!=` for string equality checking. `is` is checking for reference.

Reply:
Amended